### ISSUE TITLE
Add dl options for series and films 

### ISSUE THREAD

**Issue by gwleclerc** [matcornic/subify#3]
As Addi7ed does not have real API, it would be a great to add some options on dl command like :
1. -e, --episode
2. -s, --season
3. -n, --name
4. -ver, --version  

An other way would be to make sub-commands from apis instead of options.
In this way, each apis could have its optionsavailable using --help option like :
- subify dl addi7ed -n "IT Crowd" -s 03 -e 04 -ver "PDTV-RiVER" -l french
- subify dl addi7ed -n "chappie" -l english

**Comment by matcornic** [comment]
Thank you Gwendal for your comment. 
I just want Subify to take up the KISS principle. I think we should have only a video as an input parameter. I'd prefer parsing the video name or use the API of OpenSubtitles to get information (season, episode, etc.) about the video from the hash. Then create our own API for Addic7ed with this kind of information as an input. 

What do you think ?

**Comment by gwleclerc** [comment]
I agree with you but we need to define some pattern to parse correctly the input video. So there is some points to discuss  :
- How to determine if the input video is a film or a tv show ?
- According to the web conventions seasons and episodes should be parse using a pattern like this  
  /.*?(\d{1,2}).*?(\d{1,2}).*?/g with \1 as seasons et \2 as episodes
- How to determine which version to choose (there is many versions on addic7ed and there is no ranking system)
-  I found an api in js here : https://github.com/same31/addic7ed-api

**Comment by matcornic** [comment]
- Maybe with OpenSubtitles API. We could get information from the hash of the video. Actually we don't really care because we could use the search function of Addic7ed. I tried "Shameless.US.S06E04.720p.HDTV.X264-DIMENSION" in their main page and I was redirected to the right page. With a film name, I don't get redirected to the page, but it displays a page of results.
- Many patterns could be used indeed. But the simplest is using the search function of Addic7ed.
- No rank needed, we "just" have to found the right subtitle from the Team that edited the video (It's in the page). To extract this information, I created in 2 minutes an API with import.io It could work well with a little tuning : https://api.import.io/store/connector/20268375-825f-44e7-bbe6-4b04d8a36145/_query?input=webpage/url:http%3A%2F%2Fwww.addic7ed.com%2Fserie%2FShameless_%2528US%2529%2F6%2F4%2FGoing_Once%252C_Going_Twice&&_apikey=91b43c5e88094addbd5dd95d619139a88556fdfbaf3b1998f011d2973d4fc363b3805f7d5d3cd1f908fc0dcabd5159fc5f4724feda33dc243209a6233feb571aa2e988822aed743e335be6d7ac4e3e2f
  Other is solution is to search in the DOM like the JS API you found.
- Could be a start.

Keep in mind that aim is to give the best suitable subtitle, transparently by using several APIs. It really is a "best effort mode". If Subify can't parse for Addic7ed or don't find exactly the right subtitle, don't worry it will probably succeed in downloading from OpenSubtitles or SubDB.
